### ISSUE TITLE
JavaEE stub not working for Resource Leak Checker

### ISSUE THREAD

**Issue by msridhar** [typetools/checker-framework#5472]
Input:
```java
class Test {
    static void foo(javax.servlet.ServletResponse s) throws java.io.IOException {
	s.getWriter();
    }
}
```

In `JavaEE.astub` we have marked the `getWriter()` method as `@NotOwning`:

https://github.com/typetools/checker-framework/blob/master/checker/src/main/java/org/checkerframework/checker/mustcall/JavaEE.astub#L17-L17

And yet, compiling the above gives an error with 3.28.0:

```
$ ./checker-framework-3.28.0/checker/bin/javac -processor resourceleak -cp /path/to/javax.servlet-api-3.1.0.jar:. Test.java
Test.java:4: error: [required.method.not.called] @MustCall method close may not have been invoked on temp-var-1 or any of its aliases.
	s.getWriter();
	           ^
  The type of object is: java.io.PrintWriter.
  Reason for going out of scope: regular method exit
1 error
```

I used the following `javax.servlet-api-3.1.0.jar` in the classpath:

https://repo1.maven.org/maven2/javax/servlet/javax.servlet-api/3.1.0/javax.servlet-api-3.1.0.jar

**Comment by kelloggm** [comment]
There is a serious problem in `AnnotatedTypeFactory#getDeclAnnotation`, which (when called from `MustCallConsistencyAnalyzer`) is not producing the `@NotOwning` annotation from the stub file, contrary to its specification (checked via print-line debugging). I don't know what the cause of that problem is, but we should investigate further.

In the mean time, I'll open a PR that works around this problem in the short term by changing the `@NotOwning` annotations to `@MustCall({})`, delegating the work to the Must Call Checker.

**Comment by msridhar** [comment]
> There is a serious problem in `AnnotatedTypeFactory#getDeclAnnotation`, which (when called from `MustCallConsistencyAnalyzer`) is not producing the `@NotOwning` annotation from the stub file, contrary to its specification (checked via print-line debugging). I don't know what the cause of that problem is, but we should investigate further.
> 
> In the mean time, I'll open a PR that works around this problem in the short term by changing the `@NotOwning` annotations to `@MustCall({})`, delegating the work to the Must Call Checker.

Is there any chance we could fix the `AnnotatedTypeFactory#getDeclAnnotation` method instead of doing a workaround?  My concern is that I'm not really sure where else the RLC is currently relying on reading annotations from stub files and it is not working.  It's possible you were planning to do that and the workaround is just a super-fast temporary fix.

**Comment by kelloggm** [comment]
> It's possible you were planning to do that and the workaround is just a
super-fast temporary fix.

That's correct, but I didn't have time to look into that this morning. It
might also require expertise from either Mike or Suzanne, too, because it's
probably a framework level issue. I opened the workaround PR to unblock you.
